### DISCUSSION TITLE
Can the Brewfile & brew bundle install track that certain packages were installed without dependencies?

### DISCUSSION THREAD

**Discussion by jjt** [Homebrew/discussions#5350] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.21
ORIGIN: https://github.com/Homebrew/brew
HEAD: 82c2e743a5bcea725f9ca1429e3e21c3088ff904
Last commit: 4 days ago
Core tap JSON: 09 May 17:57 UTC
Core cask tap JSON: 09 May 17:57 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.40.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.3
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  python-click
  python-mutagen
  python-brotli

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
  python-dateutil
  python-typing-extensions

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-versions
  Homebrew/homebrew-cask-drivers
Untap them with `brew untap`.

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  nss
```

### Description of issue

I have a package (prettier) that has a dependency of node. I manage node with a different tool and _do not_ want brew to install node.

I store my Brewfile in my dotfiles repo, and use it to spin up new machines. Every time I run brew install/uninstall/etc, I dump a new Brewfile and commit it to my dotfiles repo. I would like to be able to just `brew bundle install` on a new system.

I can `brew install prettier --ignore-dependencies`, which is fine. But there is no indication of `--ignore-dependencies` in the resulting `bundle dump` output (the line is just `brew "prettier"`). 

This makes it a non-starter for me to install any package without their deps, since `brew bundle install` on a new system would install deps.

**Comment by SMillerDev** [comment]
Installing brew packages without dependencies is not really a supported case for Homebrew. The option is there, but there are no easy ways to do this in a bundle on purpose

**Reply by jjt** [reply]
> The option is there, but there are no easy ways to do this in a bundle on purpose

"The option" meaning just `--ignore-dependencies` on installing packages, or is there some non-easy way to accomplish what I'm after within brew?

I could solve my usecase in userland via explicitly filtering out packages when dumping, (`brew bundle dump --file=- | sed -E '/.*(foo|bar|baz)/d' >! $PATH_TO_BREWFILE`), and adding corresponding `brew install foo --ignore-dependencies` etc. lines to my dotfiles install.sh, but that feels brittle.

**Comment by rrotter** [comment]
To skip installation of dependencies, add this to your Brewfile:
`brew "prettier", args: ['ignore-dependencies']`

Alternatively, you could do this to leave node unlinked:
`brew "node", link: false`

The former would require an inversion of your workflow (edit Brewfile by hand, install packages w/ `brew bundle --global`). The latter would still install node but it wouldn't show on your path. The advantage here is that `brew dump` does faithfully reproduce the linked/unlinked status of your installed packages.
